src/lang/tscript-lib/core.ts

```diff
@@ -957,43 +957,60 @@ export const core = {
 		},
 		alert: function (text) {
 			let s = TScript.toString.call(this, text);
-			if (!this.service.documentation_mode && this.service.alert)
-				this.service.alert(s);
+			if (!this.service.documentation_mode && this.service.alert) {
+				this.status = "dialog";
```

Comment:
```suggestion
				this.status = "dialog";
				this.service.statechanged?.(false);
```

You probably need to call `statechanged`, as it's currently not showing any state change when opening a dialog.